ceci/stage.py

```diff
@@ -148,6 +148,10 @@ def get_aliased_tag(self, tag):
     def run(self):  # pragma: no cover
         """Run the stage and return the execution status"""
         raise NotImplementedError("run")
+    
+    def validate(self):
+        """Check that the inputs actually have the data needed for execution"""
```

Comment:
Maybe explain that this is called just before the `run` method? here.

Reply:
agreed will add